api/net/Areas/Services/Controllers/DataSourceController.cs

```diff
@@ -0,0 +1,114 @@
+public class DataSourceController : ControllerBase
+{
+    #region Variables
+    private readonly IDataSourceService _serviceDataSource;
+    private readonly IScheduleService _serviceSchedule;
+    private readonly JsonSerializerOptions _serializerOptions;
+    #endregion
+
+    #region Constructors
+    /// <summary>
+    /// Creates a new instance of a DataSourceController object, initializes with specified parameters.
+    /// </summary>
+    /// <param name="serviceDataSource"></param>
+    /// <param name="serviceSchedule"></param>
+    /// <param name="serializerOptions"></param>
+    public DataSourceController(IDataSourceService serviceDataSource, IScheduleService serviceSchedule, IOptions<JsonSerializerOptions> serializerOptions)
+    {
+        _serviceDataSource = serviceDataSource;
+        _serviceSchedule = serviceSchedule;
+        _serializerOptions = serializerOptions.Value;
+    }
+    #endregion
+
+    #region Endpoints
+    /// <summary>
+    /// Find a page of content for the specified query filter.
+    /// </summary>
+    /// <returns></returns>
+    [HttpGet("{code}")]
+    [Produces("application/json")]
+    [ProducesResponseType(typeof(DataSourceModel), (int)HttpStatusCode.OK)]
+    [ProducesResponseType((int)HttpStatusCode.NoContent)]
+    [SwaggerOperation(Tags = new[] { "DataSource" })]
+    public IActionResult FindByCode(string code)
+    {
+        var result = _serviceDataSource.FindByCode(code);
+        if (result == null) return new NoContentResult();
+        return new JsonResult(new DataSourceModel(result, _serializerOptions));
+    }
+
+    /// <summary>
+    /// Find a page of content for the specified query filter.
+    /// </summary>
+    /// <returns></returns>
+    [HttpGet("for/media/type/{mediaTypeName}")]
+    [Produces("application/json")]
+    [ProducesResponseType(typeof(IEnumerable<DataSourceModel>), (int)HttpStatusCode.OK)]
+    [SwaggerOperation(Tags = new[] { "DataSource" })]
+    public IActionResult FindByMediaType(string mediaTypeName)
+    {
+        var result = _serviceDataSource.FindByMediaType(mediaTypeName);
+        return new JsonResult(result.Select(ds => new DataSourceModel(ds, _serializerOptions)));
+    }
+
+    /// <summary>
+    /// Update data-source in database.
+    /// </summary>
+    /// <param name="model"></param>
+    /// <returns></returns>
+    [HttpPut("{id:int}")]
+    [Produces("application/json")]
+    [ProducesResponseType(typeof(DataSourceModel), (int)HttpStatusCode.OK)]
+    [ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.BadRequest)]
+    [SwaggerOperation(Tags = new[] { "DataSource" })]
+    public IActionResult Update(DataSourceModel model)
+    {
+        var result = _serviceDataSource.Update(model.ToEntity(_serializerOptions));
+
+        // Update any schedules that have changed.
+        // TODO: This is a bad design, services should not be changing the schedule.
+        model.DataSourceSchedules.ForEach(dss =>
```

Comment:
I've added this to support the current implementation of updating the schedule with the next `RunOn` time.  This isn't ideal, but for now I hope it works as it was implemented.